### ISSUE TITLE
[BUG]: Serialisation error for class with nullable struct with child property

### ISSUE THREAD

**Issue by paulengineer** [aloneguid/parquet-dotnet#495]
### Library Version

4.23.4

### OS

Windows

### OS Architecture

64 bit

### How to reproduce?

1. Add test below (possibly to ParquetSerializerTest.cs ?) and run

### Failing test

```csharp
private struct StructWithIntProp {
    public int id { get; set; }
}
private class ClassWithNullableCustomStruct {
    public StructWithIntProp? nullableStruct { get; set; }
}

[Fact]
public async Task Class_With_Nullable_Struct() {
    var data = new List<ClassWithNullableCustomStruct> {
        new ClassWithNullableCustomStruct() {
            nullableStruct = null
        }
    };

    using var ms = new MemoryStream();
    await ParquetSerializer.SerializeAsync(data, ms);

    ms.Position = 0;
    IList<ClassWithNullableCustomStruct> data2 = await ParquetSerializer.DeserializeAsync<ClassWithNullableCustomStruct>(ms);

    Assert.Equivalent(data2, data);
}
```

**Comment by aloneguid** [comment]
error:

```
Message: 
  System.ApplicationException : failed to serialise data column 'NullableStruct/Value/Id'
  ---- System.InvalidOperationException : Nullable object must have a value.

Stack Trace: 
  ParquetSerializer.SerializeRowGroupAsync[T](ParquetWriter writer, Striper`1 striper, IEnumerable`1 objectInstances, CancellationToken cancellationToken) line 43
  ParquetSerializer.SerializeAsync[T](IEnumerable`1 objectInstances, Stream destination, ParquetSerializerOptions options, CancellationToken cancellationToken) line 118
  ParquetSerializer.SerializeAsync[T](IEnumerable`1 objectInstances, Stream destination, ParquetSerializerOptions options, CancellationToken cancellationToken) line 120
  ParquetSerializerTest.Class_With_Nullable_Struct() line 1071
  --- End of stack trace from previous location ---
  ----- Inner Stack Trace -----
  lambda_method38(Closure, DataField, IEnumerable`1)
  ParquetSerializer.SerializeRowGroupAsync[T](ParquetWriter writer, Striper`1 striper, IEnumerable`1 objectInstances, CancellationToken cancellationToken) line 39

```

**Comment by aloneguid** [comment]
@paulengineer of course as a workaround you can use class instead of struct.

**Comment by aloneguid** [comment]
fix included in 5.0.2-pre.2

**Comment by paulengineer** [comment]
Thank-you.
